packages/block-library/src/cover/edit/index.js

```diff
@@ -59,6 +61,82 @@ function getInnerBlocksTemplate( attributes ) {
 	];
 }
 
+/**
+ * Computes the average color of an image.
+ * @param {string} url
+ * @return {Promise<string>} average color in hex
+ */
+async function computeAverageColor( url ) {
+	function retrieveFastAverageColor() {
+		if ( ! retrieveFastAverageColor.fastAverageColor ) {
+			retrieveFastAverageColor.fastAverageColor = new FastAverageColor();
+		}
+		return retrieveFastAverageColor.fastAverageColor;
+	}
+
+	// making the default color rgb for compat with FAC
+	const { r, g, b, a } = colord( DEFAULT_AVERAGE_COLOR ).toRgb();
+
+	try {
+		const imgCrossOrigin = applyFilters(
+			'media.crossOrigin',
+			undefined,
+			url
+		);
+		const color = await retrieveFastAverageColor().getColorAsync( url, {
+			// The default color is white, which is the color
+			// that is returned if there's an error.
+			// colord returns alpga 0-1, FAC needs 0-255
+			defaultColor: [ r, g, b, a * 255 ],
+			// Errors that come up don't reject the promise,
+			// so error logging has to be silenced
+			// with this option.
+			silent: process.env.NODE_ENV === 'production',
+			crossOrigin: imgCrossOrigin,
+		} );
+		return color.hex;
+	} catch ( error ) {
+		// If there's an error return the fallback color.
+		return '#FFFFFF';
+	}
+}
+
+/**
+ * Computes if the color combination of the overlay
+ * and background color is dark.
+ * @param {number} dimRatio
+ * @param {string} overlayColor
+ * @param {string} backgroundColor
+ * @return {boolean} isDark true if the color
+ * 									 combination composite result is dark
+ */
+function computeIsDark( dimRatio, overlayColor, backgroundColor ) {
+	/**
+	 * Performs a Porter Duff composite source over operation on two rgba colors.
+	 *
+	 * @see https://www.w3.org/TR/compositing-1/#porterduffcompositingoperators_srcover
+	 *
+	 * @param {import('colord').RgbaColor} source Source color.
+	 * @param {import('colord').RgbaColor} dest   Destination color.
+	 * @return {import('colord').RgbaColor} Composite color.
+	 */
+	function compositeSourceOver( source, dest ) {
+		return {
+			r: source.r * source.a + dest.r * dest.a * ( 1 - source.a ),
+			g: source.g * source.a + dest.g * dest.a * ( 1 - source.a ),
+			b: source.b * source.a + dest.b * dest.a * ( 1 - source.a ),
+			a: source.a + dest.a * ( 1 - source.a ),
+		};
+	}
```

Comment:
This is a pure function, so I don't see any reason that we have to re-create it every time we call `computeIsDark`. This should be a top-level function like before.